src/ship/Propulsion.cpp

```diff
@@ -327,16 +320,29 @@ bool Propulsion::AIMatchVel(const vector3d &vel, const vector3d &powerLimit)
 // returns true if this can be done in a single timestep
 bool Propulsion::AIChangeVelBy(const vector3d &diffvel, const vector3d &powerLimit)
 {
-	// counter external forces
-	vector3d extf = m_dBody->GetExternalForce() * (Pi::game->GetTimeStep() / m_dBody->GetMass());
-	vector3d diffvel2 = diffvel - extf * m_dBody->GetOrient();
+	float frameTimestep = Pi::game->GetTimeStep();
+
+	// external force, acceleration and velocity
+	// acceleration: F = ma -> a = F / m
+	vector3d extf = m_dBody->GetExternalForce() * m_dBody->GetOrient();
+	vector3d exta = extf / m_dBody->GetMass();
+	vector3d extdv = exta * frameTimestep;
+
+	// required internal velocity change (should extdv be the opposite direction?)
+	vector3d intdvGoal = diffvel - extdv;
+
+	// internal force and acceleration
+	vector3d intf = GetThrust(intdvGoal);
+	vector3d inta = intf / m_dBody->GetMass();
+	vector3d intdv = inta * frameTimestep;
+
+	vector3d intdvSmooth = intdv * 60.0;
```

Comment:
What is the purpose of the magic number `60` here? Is it related to the framerate or just an arbitrary scalar?

Reply:
Just an arbitrary scalar, it smoothens the calculated force by 1 second of gameplay (but could be anything)